src/transport/SessionHandle.cpp

```diff
@@ -23,12 +23,33 @@ namespace chip {
 
 using namespace Transport;
 
+using AuthMode          = Access::AuthMode;
 using SubjectDescriptor = Access::SubjectDescriptor;
 
 SubjectDescriptor SessionHandle::GetSubjectDescriptor() const
 {
-    SubjectDescriptor subjectDescriptor = { .fabricIndex = mFabric };
-    // TODO: fill subject descriptor with proper fields
+    SubjectDescriptor subjectDescriptor;
+    if (IsSecure())
+    {
+        if (IsOperationalNodeId(mPeerNodeId))
+        {
+            subjectDescriptor.authMode    = AuthMode::kCase;
+            subjectDescriptor.subject     = mPeerNodeId;
+            subjectDescriptor.fabricIndex = mFabric;
+            // TODO(#10243): add CATs
+        }
+        else if (IsPAKEKeyId(mPeerNodeId))
+        {
+            subjectDescriptor.authMode = AuthMode::kPase;
+            subjectDescriptor.subject  = mPeerNodeId;
+            // TODO(#10242): PASE *can* have fabric in some situations
+        }
+        else if (mGroupId.HasValue())
+        {
+            subjectDescriptor.authMode = AuthMode::kGroup;
+            subjectDescriptor.subject  = kMinGroupNodeId | mGroupId.Value();
```

Comment:
Shouldn't this operation be implemented as a function somewhere?  Something like `NodeIdFromGroupId` or `GroupIdToNodeId` or some such?

Reply:
That would make sense to me.  It'lkl need to include GroupId.h, but that seems fine.